sema/subyt/__main__.py

```diff
@@ -0,0 +1,218 @@
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="FILE|PATTERN",  # meaning of the argument
+        action="store",
+        help="Specifies where to write the output, can use {uritemplate}.",
+    )
+    parser.add_argument(
+        "-f",
+        "--force",
+        default=False,
+        action="store_true",
+        help=(
+            "Force writing output, do not check "
+            "if output files already exist."
+        ),
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        metavar=" (no-)it(eration), (no-)ig(norecase), (no-)fl(atten) ",
+        action="store",
+        help="""Modifies the mode of iteration:
+                1. it (default) vs. no-it: apply template for each
+                                                iterated row in the input set
+                                           vs. apply it only once once for
+                                                the complete input set;
+                2. ig vs. no-ig: to be implemented;
+                3. fl vs. no-fl: to be implemented.""",
+    )
+    parser.add_argument(
+        "-r",
+        "--allow-repeated-sink-paths",
+        default=False,
+        action="store_true",
+        help=("Allow repeated sink paths in case of duplicated data items."),
+    )
+    parser.add_argument(
+        "-c",
+        "--conditional",
+        default=False,
+        action="store_true",
+        help=("Execute only when input has been updated. Abort otherwise."),
+    )
+    return parser
+
+
+def make_service(args: argparse.Namespace) -> Generator:
+    template_folder = args.templates
+    return JinjaBasedGenerator(template_folder)
+
+
+def make_sources(args: argparse.Namespace) -> dict:
+    inputs = dict()
+    if args.input is not None:
+        inputs["_"] = SourceFactory.make_source(args.input)
+    if args.set is not None:
+        for [key, file_name] in args.set:
+            inputs[key] = SourceFactory.make_source(file_name)
+    return inputs
+
+
+def make_sink(args: argparse.Namespace) -> Sink:
+    return SinkFactory.make_sink(
+        args.output, args.force, args.allow_repeated_sink_paths
+    )
+
+
+def vars_to_dict(vars: list) -> dict:
+    if not vars:
+        return None
+    # else
+    return {name: value for [name, value] in vars}
+
+
+def enable_logging(args: argparse.Namespace):
+    if args.logconf is None:
+        return
+    import yaml  # conditional dependency -- we only need this (for now)
+
+    # when logconf needs to be read
+
+    with open(args.logconf, "r") as yml_logconf:
+        logging.config.dictConfig(
+            yaml.load(yml_logconf, Loader=yaml.SafeLoader)
+        )
+    log.info(f"Logging enabled according to config in {args.logconf}")
+
+
+def main():
+    """
+    The main entry point to this module.
+    """
+    args = get_arg_parser().parse_args()
+
+    enable_logging(args)
+    service = make_service(args)
```

Comment:
precursor of the new style "service" interface introduced by sembench
-- maybe we should consider supporting that way of calling the core process with string-arguments only already?

Reply:
taken up in issue #71